cmd/skyeye/main.go

```diff
@@ -53,48 +55,112 @@ var (
 )
 
 func init() {
+	skyeye.Flags().StringVar(&configPath, "config-path", ".", "Path to a config file e.g. '/home/user/xyz'. It is looking for a file called skyeye-config.yaml")
+
+	viper.SetConfigName("skyeye-config")
+	viper.SetConfigType("yaml")
+	viper.AddConfigPath(configPath)
+	if err := viper.ReadInConfig(); err != nil {
+		log.Error().Err(err).Msg("Unable to read config yaml file")
+	}
+
 	// Logging
 	logLevelFlag := NewEnum(&logLevel, "Level", "info", "error", "warn", "info", "debug", "trace")
 	skyeye.Flags().Var(logLevelFlag, "log-level", "Log level (error, warn, info, debug, trace)")
+	viper.BindPFlag("log-level", skyeye.Flags().Lookup("log-level"))
+	viper.SetDefault("log-level", "info")
```

Comment:
Is there a way to avoid the duplication of the default settings here? Maybe a helper function?

Reply:
I tried to do a helper function like this:

```
func bindFlag[T any](cmd *cobra.Command, viperKey string, defaultValue T, usage string, flagSetter func(flag *pflag.FlagSet)) {
	flagSetter(cmd.Flags())
	viper.BindPFlag(viperKey, cmd.Flags().Lookup(viperKey))
	viper.SetDefault(viperKey, defaultValue)
}
```

However this still results in a function call like this which isn't really shorter and much less readable code wise:
```
// Logging
bindFlag(skyeye, "log-level", "info", "Log level (error, warn, info, debug, trace)", func(flags *pflag.FlagSet) {
	logLevelFlag := NewEnum(&logLevel, "Level", "info", "error", "warn", "info", "debug", "trace")
	flags.Var(logLevelFlag, "log-level", "Log level (error, warn, info, debug, trace)")
})
```

Let me know if you have a better idea :)